e2e/email-sign-up.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { test, expect } from '@playwright/test';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/email-sign-up');
+});
+
+test.describe('Email sign-up page', () => {
+  test('The page renders with correct title', async ({ page }) => {
+    await expect(page).toHaveTitle(`Email Sign Up | freeCodeCamp.org`);
+  });
+
+  test('The page renders the correct content', async ({ page }) => {
+    await expect(
+      page.getByText(
+        'freeCodeCamp is a proven path to your first software developer job.'
+      )
+    ).toBeVisible();
+  });
+
+  test('The sign up button renders and functions', async ({ page }) => {
+    const button = page.getByTestId('email-sign-up-button');
+
+    await expect(button).toBeVisible();
+
+    await button.click();
+
+    await expect(page).toHaveURL(/.*\/learn\/?$/);
+  });
```

Comment:
```suggestion
  test('should render correctly', async ({ page }) => {
    await expect(page).toHaveTitle(`Email Sign Up | freeCodeCamp.org`);
    await expect(
      page.getByText(
        'freeCodeCamp is a proven path to your first software developer job.'
      )
    ).toBeVisible();
    const button = page.getByTestId('email-sign-up-button');
    await expect(button).toBeVisible();
    await button.click();
    await expect(page).toHaveURL(/.*\/learn\/?$/);
  });
```
We can probably combine all of these into a single rendering test.